cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -338,6 +338,35 @@ class string_scalar : public scalar {
   rmm::device_buffer _data{};  ///< device memory containing the string
 };
 
+/**
+ * @brief An owning class to represent a list in device memory
+ */
+class list_scalar : public scalar {
```

Comment:
Is this really needed? Seems an element of a list-view is a column of something. Will this scalar be a wrapper of a column so it can be used as a  scalar in operations like `fill()`?
Not all columns should require a scalar. For example, I don't expect to need a scalar class for dictionary since it should work with scalar of whatever the key type should be.

Reply:
Right, these are good questions.  A scalar for a list column is indeed a column.   Potentially further complicating things is the fact that it could also have children.   It does seem possible that no one would ever want to manipulate one of these things as a scalar.   @jlowe  ?

Reply:
In Spark SQL it is possible to create scalars of complex types, so yes, we do need support for nested type scalars.

The `fill` example is useful here.  Worst-case if libcudf doesn't support operating on a scalar type for some operation then we have to resort to using `fill` to emulate it.  Therefore we need a way to use nested types with `fill` because users can specify nested type literals in their queries.

This revisits the "are scalars really just a 1-row column" discussion.  As long as there's a way to indicate the semantics are scalar-like instead of column-like I don't care too much either way.  I'm sure others have stronger opinions.

Reply:
I guess my point is that the current `list_scalar` defined here does not do anything and does not appear to have any data or have any purpose. I would've expected it to contain a column perhaps. So maybe it is just not fully baked and should not be added until we can figure out what it is for.
This is independent of the _scalar is really just a 1-row column_ discussion.  I don't picture `list_view` column type being used with any existing scalar API. There may be current scalar-ish APIs that we need to overload to accept a column instead of a scalar. The [fill](https://github.com/rapidsai/cudf/blob/branch-0.14/cpp/include/cudf/filling.hpp) APIs are a good example where we may want to overload them to accept a column parameter in place of a scalar.

Reply:
Agree it may be too early to add the type until it has a use.  From my perspective, `fill` is the point where we need to decide how it's going to be done.  If `fill` is supposed to only operate with a scalar then there needs to be a `scalar` type that can be constructed for nested types and passed as the argument with which to fill.  If we're going to allow `fill` to take a column as well then we're opening the door for scalars to also be columns which why I see it as a related issue.

Reply:
Ok, so consensus is :  remove for now?